src/main/java/org/opensearch/security/securityconf/ConfigModelV7.java

```diff
@@ -484,6 +485,69 @@ public boolean impliesClusterPermissionPermission(String action) {
             return roles.stream().filter(r -> r.impliesClusterPermission(action)).count() > 0;
         }
 
+        /**
+         * Checks if the route is accessible via legacy naming convention.
+         * We check against cluster permissions because the legacy convention for cluster permissions map 1-1 with a transport
+         * action call initiated via REST API handler. Hence we use the same to allow/block request forwarding to ext.
+         * This ensures backwards-compatibility
+         *
+         *
+         * NOTE: THIS CHECK WILL BE REMOVED ONCE ALL ACTIONS HAVE BEEN MIGRATED TO THE NEW CONVENTION
+         *
+         * E.g For ext `hw`, following are two possible ways actions an be defined in roles:
+         *
+         * ext_hw_greet:
+         *   reserved: true
+         *   cluster_permissions:
+         *     - 'hw:greet'
+         *
+         * legacy_hw_greet:
+         *   reserved: true
+         *   cluster_permissions:
+         *     - 'cluster:admin/opensearch/hw/greet'
+         *
+         *
+         * @param action - The action to be checked against its legacy version
+         * @return true, if a legacy version was found and validated, false otherwise
+         */
+        @Override
+        public boolean impliesLegacyPermission(String action) {
```

Comment:
Why did we need to alter the config model?  Permissions themselves aren't any different, the only difference is how the RestPrivilegeEvaluator checks them, if any match from the NamedRoutes instead of checking a single action name

Reply:
This was added to ensure that the new naming convention still matches the old roles defined. i.e. `ad:detector/info` will match against `cluster:admin/opensearch/ad/detector/info`. Without this, the new permissions will not match against legacy ones.

Reply:
This is a good integration test scenario to add

Reply:
IMO we don't need this since `NamedRoutes` support registering legacy action names as well.

Reply:
this method evaluates new permissions against legacy permission. For example, a route `GET /x` is registered with a unique name `getX` and no legacy permission actions. We can't directly evaluate using `impliesClusterPermission`, instead we would need this method to translate `getX` to a legacy name, i.e.`cluster:admin/opensearch/plugin1/getX` and then evaluated.

Reply:
We won't need this at all if we are mandating actionNames to be associated with each PR. Then we will rely on those to evaluate against the user's permissions.